ibis/backends/tests/test_export.py

```diff
@@ -0,0 +1,132 @@
+import sys
+
+import pyarrow as pa
+import pytest
+from pytest import param
+
+
+class PackageDiscarder:
+    def __init__(self):
+        self.pkgnames = []
+
+    def find_spec(self, fullname, path, target=None):
+        if fullname in self.pkgnames:
+            raise ImportError()
+
+
+@pytest.fixture
+def no_pyarrow(backend):
+    _pyarrow = sys.modules.pop('pyarrow', None)
+    d = PackageDiscarder()
+    d.pkgnames.append('pyarrow')
+    sys.meta_path.insert(0, d)
+    yield
+    sys.meta_path.remove(d)
+    if _pyarrow is not None:
+        sys.modules["pyarrow"] = _pyarrow
+
+
+limit_no_limit = [
+    param(
+        42,
+        id='limit',
+        marks=[
+            pytest.mark.notimpl(
+                [
+                    # limit not implemented for pandas backend execution
+                    "pandas",
+                    "dask",
+                    "clickhouse",
+                    "impala",
+                    "pyspark",
+                    "datafusion",
+                ]
+            ),
+        ],
+    ),
+    param(
+        None,
+        id='nolimit',
+        marks=[
+            pytest.mark.notimpl(
+                [
+                    "dask",
+                    "clickhouse",
+                    "impala",
+                    "pyspark",
+                ]
+            ),
+        ],
+    ),
+]
+
+
+@pytest.mark.parametrize("limit", limit_no_limit)
+def test_to_pyarrow_batches(limit, backend, awards_players):
```

Comment:
@gforsyth could you please add tests for expressions with different shape, like `table.column.to_pyarrow()` and `table.column.sum().to_pyarrow()`?

Reply:
Hmmm, `to_pyarrow`/`to_pyarrow_batches`/`to_pylist`/`to_pylist_batches` all feel weird to me to be on a `Scalar` expr. Should we instead add those only on `Column` and `Table` expressions?

Reply:
I moved `to_pyarrow` and `to_pyarrow_batches` to `Column` and `Table` -- also added tests for both table and column exports.

It's still possible to call `to_pyarrow` from the backend directly on a scalar -- I wonder if we should be adding in (basic) dimension information, like a `ndims` attribute to expressions to make this easier to dispatch on

Reply:
I think it's strange to not define `to_pyarrow` for scalars. It makes downstream integration and use of the API difficult because now such consumers have to build additional code to handle the scalar case uniformly. There are pyarrow scalar objects that would make sense here. Can't we handle that for users?

Reply:
Ok, i can see a two ways of handling this (there are others, but they seem less self-consistent):

1.  Everything is a table
Table -> pa.Table
Column -> pa.Table with only one column
Scalar -> pa.Table with only one column and one row
^^^ this is probably wacky

2. Things go to their corresponding shape
Table -> pa.Table
Column -> pa.Array
Scalar -> pa.Scalar

This is more consistent, but downstream users will probably still need additional code to handle the output

Reply:
I am also aware this PR has been open for a while so please do not consider my comments blocking.

Reply:
1 makes the most sense to me for Tables/columns, but for scalars (if we handle those) I think the scalar should be returned directly. Idk though.

Phillip's argument about API consistency makes sense to me for `to_pyarrow`, especially since there are `pyarrow` types for scalars. `to_pylist` for scalars still feels a bit off (if the scalar type is an `Array`/`Struct`/`Map` type are we converting to a list, or is a list just the outer collection for table/column types?). The `to_*_batches` methods also feel off to me to define on scalars, Phillip are you fine dropping those from scalar types?

Reply:
I'd favor the second case for consistency, the user can easily convert the pyarrow scalar to a python scalar using `pa_scalar.as_py()`

Reply:
@kszucs do you think a `Column` should map to a `pa.Table` with a single column, or a `pa.Array` type?

Reply:
IMO the latter, currently we return with a pandas series for a column expression